components/kyma-environment-broker/cmd/subscriptioncleanup/cloudprovider/aws.go

```diff
@@ -1,18 +1,126 @@
 package cloudprovider
 
+import (
+	"context"
+	"errors"
+	"log"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+)
+
+const defaultRegion string = "eu-central-1"
+
 type awsResourceCleaner struct {
+	ec2Client   ec2.Client
+	credentials awsCredentialsConfig
 }
 
-type awsConfig struct {
+type awsCredentialsConfig struct {
 	accessKeyID     string
 	secretAccessKey string
 }
 
 func NewAwsResourcesCleaner(secretData map[string][]byte) (ResourceCleaner, error) {
-	return awsResourceCleaner{}, nil
+	awsResourceCleaner := awsResourceCleaner{}
+	awsConfig, err := awsResourceCleaner.toAwsConfig(secretData)
+	if err != nil {
+		return nil, err
+	}
+
+	awsResourceCleaner.credentials = awsConfig
+	awsEC2Client, err := awsResourceCleaner.newAwsEC2Client(awsConfig, defaultRegion)
```

Comment:
I think since the client is per-region, the cleaner struct shouldn't have a (global) attribute for it, and it's nto necessary to initialize a client with a "default" region, which is never being used during code execution.

I would just-in-time initialize the client when looping through the regions below, and pass the client as a parameter of  `deleteVolumes(client)`